opencensus/trace/exporters/jaeger_exporter.py

```diff
@@ -0,0 +1,451 @@
+from thrift.protocol import TCompactProtocol, TBinaryProtocol
+
+from opencensus.trace import span_data, link
+from opencensus.trace.exporters import base
+from opencensus.trace.exporters.transports import sync
+from opencensus.trace.exporters.gen.jaeger import jaeger
+from opencensus.trace.exporters.gen.jaeger import agent
+
+DEFAULT_HOST_NAME = 'localhost'
+DEFAULT_AGENT_PORT = 6931
+DEFAULT_ENDPOINT = '/api/traces'
+
+ISO_DATETIME_REGEX = '%Y-%m-%dT%H:%M:%S.%fZ'
+UDP_PACKET_MAX_LENGTH = 65000
+
+logging = logging.getLogger(__name__)
+
+
+class JaegerExporter(base.Exporter):
+    """Exports the spans to Jaeger.
+
+    :type service_name: str
+    :param service_name: Service that logged an annotation in a trace.
+                         Classifier when query for spans.
+
+    :type host_name: str
+    :param host_name: (Optional) The host name of the Jaeger HTTP Thrift.
+
+    :type port: int
+    :param port: (Optional) The port of the Jaeger HTTP Thrift.
+
+    :type endpoint: str
+    :param port: (Optional) The endpoint of the Jaeger HTTP Thrift .
+
+    :type agent_host_name: str
+    :param agent_host_name: (Optional) The host name of the Jaeger-Agent.
+
+    :type agent_port: int
+    :param agent_port: (Optional) The port of the Jaeger-Agent.
+
+    :type username: str
+    :param username: (Optional) The user name of the Basic Auth
+                     if authentication is required.
+
+    :type password: str
+    :param password: (Optional) The password of the Basic Auth
+                     if authentication is required.
+
+    :type transport: :class:`type`
+    :param transport: Class for creating new transport objects. It should
+                      extend from the base :class:`.Transport` type and
+                      implement :meth:`.Transport.export`. Defaults to
+                      :class:`.SyncTransport`. The other option is
+                      :class:`.BackgroundThreadTransport`.
+    """
+
+    def __init__(
+            self,
+            service_name='my_service',
+            host_name=None,
+            port=None,
+            endpoint='',
+            agent_host_name=DEFAULT_HOST_NAME,
+            agent_port=DEFAULT_AGENT_PORT,
+            agent_endpoint=DEFAULT_ENDPOINT,
+            username='',
+            password='',
+            transport=sync.SyncTransport):
+        self.transport = transport(self)
+        self.service_name = service_name
+        self.host_name = host_name
+        self.agent_host_name = agent_host_name
+        self.port = port
+        self.agent_port = agent_port
+        self.endpoint = endpoint
+        self.username = username
+        self.password = password
+        self.agent_client = self._agent_client
+        self.collector = self._collector
+
+    @property
+    def _agent_client(self):
+        return AgentClientUDP(
+            host_name=self.agent_host_name,
+            port=self.agent_port)
+
+    @property
+    def _collector(self):
```

Comment:
You may want to cache this so you don't need to make a new thrift client for each batch.

Reply:
Agreed.